evm-contracts/src/v0.6-ovm/FluxAggregator.sol

```diff
@@ -0,0 +1,1053 @@
+
+    emit OracleAdminUpdateRequested(_oracle, msg.sender, _newAdmin);
+  }
+
+  /**
+   * @notice accept the admin address transfer for an oracle
+   * @param _oracle is the address of the oracle whose admin is being transferred
+   */
+  function acceptAdmin(address _oracle)
+    external
+  {
+    require(oracles[_oracle].pendingAdmin == msg.sender, "only callable by pending admin");
+    oracles[_oracle].pendingAdmin = address(0);
+    oracles[_oracle].admin = msg.sender;
+
+    emit OracleAdminUpdated(_oracle, msg.sender);
+  }
+
+  /**
+   * @notice allows non-oracles to request a new round
+   */
+  function requestNewRound()
+    external
+    returns (uint80)
+  {
+    require(requesters[msg.sender].authorized, "not authorized requester");
+
+    uint32 current = reportingRoundId;
+    require(rounds[current].updatedAt > 0 || timedOut(current), "prev round must be supersedable");
+
+    uint32 newRoundId = current.add(1);
+    requesterInitializeNewRound(newRoundId);
+    return newRoundId;
+  }
+
+  /**
+   * @notice allows the owner to specify new non-oracles to start new rounds
+   * @param _requester is the address to set permissions for
+   * @param _authorized is a boolean specifying whether they can start new rounds or not
+   * @param _delay is the number of rounds the requester must wait before starting another round
+   */
+  function setRequesterPermissions(address _requester, bool _authorized, uint32 _delay)
+    external
+    onlyOwner()
+  {
+    if (requesters[_requester].authorized == _authorized) return;
+
+    if (_authorized) {
+      requesters[_requester].authorized = _authorized;
+      requesters[_requester].delay = _delay;
+    } else {
+      delete requesters[_requester];
+    }
+
+    emit RequesterPermissionsSet(_requester, _authorized, _delay);
+  }
+
+  /**
+   * @notice called through LINK's transferAndCall to update available funds
+   * in the same transaction as the funds were transferred to the aggregator
+   * @param _data is mostly ignored. It is checked for length, to be sure
+   * nothing strange is passed in.
+   */
+  function onTokenTransfer(address, uint256, bytes calldata _data)
+    external
+  {
+    require(_data.length == 0, "transfer doesn't accept calldata");
+    updateAvailableFunds();
+  }
+
+  /**
+   * @notice a method to provide all current info oracles need. Intended only
+   * only to be callable by oracles. Not for use by contracts to read state.
+   * @param _oracle the address to look up information for.
+   */
+  function oracleRoundState(address _oracle, uint32 _queriedRoundId)
+    external
+    view
+    returns (
+      bool _eligibleToSubmit,
+      uint32 _roundId,
+      int256 _latestSubmission,
+      uint64 _startedAt,
+      uint64 _timeout,
+      uint128 _availableFunds,
+      uint8 _oracleCount,
+      uint128 _paymentAmount
+    )
+  {
+    // OVM CHANGE: tx.origin not supported
+    // require(msg.sender == tx.origin, "off-chain reading only");
```

Comment:
```suggestion
    require(msg.sender == address(0), "off-chain reading only");
```

Reply:
Yes, this is changed in the next PR.

This PR is just C/P all contracts as is while commenting out unsupported opcodes so that compiler works.